### PR TITLE
Disallow control characters in literals

### PR DESCRIPTION
**Pull request by rossberg-chromium**
Addresses #142.

### CODE REVIEW

File: ml-proto/host/lexer.mll [hunk 1]
```diff
@@ -88,7 +88,7 @@ let letter = ['a'-'z''A'-'Z']
 let symbol = ['+''-''*''/''\\''^''~''=''<''>''!''?''@''#''$''%''&''|'':''`''.']
 let tick = '\''
 let escape = ['n''t''\\''\'''\"']
-let character = [^'"''\\''\n'] | '\\'escape | '\\'hexdigit hexdigit
+let character = [^'"''\\''\x00'-'\x1f'] | '\\'escape | '\\'hexdigit hexdigit
```
Comment (sunfishcode) [inline ml-proto/host/lexer.mll]: This (an below) should also exclude \x7f, the DEL character.

**Comment by sunfishcode** [comment]
sgtm. Is it possible to write a test confirming that control characters are rejected?

**Comment by jfbastien** [comment]
For the record, I think we'll want something slightly different as I mention in https://github.com/WebAssembly/spec/issues/142#issuecomment-148754396

This change shouldn't be blocked on it, though.

File: ml-proto/host/lexer.mll (hunk 1 above)
Comment (rossberg-chromium) [inline ml-proto/host/lexer.mll]: Done.

**Comment by rossberg-chromium** [comment]
> Is it possible to write a test confirming that control characters are rejected?

Difficult. We don't have any infrastructure for testing syntax errors, which could only work outside the script language. Not sure it's worth the effort either.

**Comment by rossberg-chromium** [comment]
Addressed comment, PTAL.

**Comment by sunfishcode** [comment]
lgtm

**Pull request merged**